VideoLocker/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -144,18 +144,46 @@ public void updateMessageView(View view){
     private void initializeAdapter(){
         if (adapter == null) {
             // creating adapter just once
-            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase(), environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
+            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase(),
+                    environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
                 @Override
                 public void download(List<HasDownloadEntry> models) {
-                    downloadManager.downloadVideos(
-                            (List) models, (FragmentActivity) getActivity(), (VideoDownloadHelper.DownloadManagerCallback) getActivity());
-
+                    CourseOutlineActivity activity = (CourseOutlineActivity) getActivity();
+                    if (new PrefManager(activity, PrefManager.Pref.WIFI).getBoolean(PrefManager.Key.DOWNLOAD_ONLY_ON_WIFI, true)) {
+                        if (!NetworkUtil.isConnectedWifi(activity)) {
+                            activity.showInfoMessage(getString(R.string.wifi_off_message));
+                            return;
+                        }
+                    } else {
+                        if (AppConstants.offline_flag) {
+                            activity.showInfoMessage(getString(R.string.network_not_connected));
+                            return;
+                        }
+                    }
+                    downloadManager.downloadVideos(models, getActivity(),
+                            (VideoDownloadHelper.DownloadManagerCallback) getActivity());
                 }
 
                 @Override
                 public void download(DownloadEntry videoData) {
-                    downloadManager.downloadVideo(
-                            videoData, (FragmentActivity) getActivity(), (VideoDownloadHelper.DownloadManagerCallback) getActivity());
+                    CourseOutlineActivity activity = (CourseOutlineActivity) getActivity();
+                    if (new PrefManager(activity, PrefManager.Pref.WIFI).getBoolean(PrefManager.Key.DOWNLOAD_ONLY_ON_WIFI, true)) {
+                        if (!NetworkUtil.isConnectedWifi(activity)) {
+                            activity.showInfoMessage(getString(R.string.wifi_off_message));
```

Comment:
Since this logic is being reused in both methods, we should probably port it into a utility method.